docs/deploy/Deploy-FhirConverterService.bicep

```diff
@@ -0,0 +1,193 @@
+
+@description('Name of the container in the storage account containing custom templates. Leave blank if using default templates.')
+param templateStorageAccountContainerName string = ''
+
+@description('Minimum possible number of replicas per revision as the container app scales.')
+param minReplicas int = 0
+
+@description('Maximum possible number of replicas per revision as the container app scales.')
+param maxReplicas int = 30
+
+@description('CPU usage limit in cores.')
+param cpuLimit string = '1.0'
+
+@description('Memory usage limit in Gi.')
+param memoryLimit string = '2Gi'
+
+@description('If set to true, security will be enabled on the API endpoint.')
+param securityEnabled bool = false
+
+@description('List of audiences that the security token is intended for.')
+param securityAuthenticationAudiences array = []
+
+@description('Issuing authority of the JWT token.')
+param securityAuthenticationAuthority string = ''
+
+@description('Tag of the image to deploy.')
+param imageTag string
+
+@description('Timestamp to append to container name. Defaults to time of deployment.')
+param timestamp string = utcNow('yyyyMMddHHmmss')
+
+// Security configuration
+var securityEnabledConfigName = 'ConvertService__Security__Enabled'
+var securityAuthenticationAudiencesConfigNamePrefix = 'ConvertService__Security__Authentication__Audiences__'
+var securityAuthenticationAuthorityConfigName = 'ConvertService__Security__Authentication__Authority'
+var securityConfiguration = [
+  {
+    name: securityEnabledConfigName
+    value: string(securityEnabled)
+  }
+  {
+    name: securityAuthenticationAuthorityConfigName
+    value: securityAuthenticationAuthority
+  }
+]
+
+var securityAuthenticationAudiencesConfig = [for (audience, i) in securityAuthenticationAudiences: {
+    name: '${securityAuthenticationAudiencesConfigNamePrefix}${i}'
+    value: audience
+}]
+
+// Template hosting configuration
+var storageEnvironmentSuffix = az.environment().suffixes.storage
+var blobTemplateHostingConfigurationName = 'TemplateHosting__StorageAccountConfiguration__ContainerUrl'
+var blobTemplateHostingConfigurationValue = 'https://${templateStorageAccountName}.blob.${storageEnvironmentSuffix}/${templateStorageAccountContainerName}'
+var blobTemplateHostingConfiguration = [
+  {
+    name: blobTemplateHostingConfigurationName
+    value: blobTemplateHostingConfigurationValue
+  }
+]
+
+// Telemetry configuration
+var applicationInsightsName = '${envName}-ai'
+resource applicationInsights 'Microsoft.Insights/components@2020-02-02' existing = {
+  name: applicationInsightsName
+}
+
+resource applicationInsightsUAMI 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' existing = {
+  name: '${applicationInsightsName}-mi'
+}
+
+var appInsightsConnectionStringConfigurationName = 'ConvertService__Telemetry__AzureMonitor__ApplicationInsightsConnectionString'
+var appInsightsConnectionString = applicationInsights.properties.ConnectionString

```

Comment:
If appInsights is not deployed (I think this can be optional?), this will throw an exception.